toasty/image.py

```diff
@@ -868,12 +857,34 @@ def aspil(self):
         """
         if self._pil is not None:
             return self._pil
-        if self.mode.try_as_pil() is None:
+        if self.mode not in (ImageMode.RGB, ImageMode.RGBA):
             raise Exception(
                 f"Toasty image with mode {self.mode} cannot be converted to PIL"
             )
+
         return pil_image.fromarray(self._array)
 
+    def coerce_into_pil(self, pixel_cut_low, pixel_cut_high):
+        """Coerce the image data into a :class:`PIL.Image.Image` by converting
+        the data into an ``uint8`` RGB(A) array.
+
+        Parameters
+        ----------
+        pixel_cut_low : number
+            An value used to stretch the pixel values to the new ``uint8``
+            range (0 - 255).
+        pixel_cut_high : number
+            An value used to stretch the pixel values to the new ``uint8``
+            range (0 - 255).
+        """
+        array = np.copy(self._array)
+        array[..., :3] = (array[..., :3] - pixel_cut_low) / (
+            pixel_cut_high - pixel_cut_low
+        ) * 255 + 0.5
```

Comment:
Hmm, now that I think about this part more, I think it will do the wrong thing in some important circumstances.

First. we should always use `self.asarray()` instead of directly accessing `self._array`.

But second, if `self._pil` is not None, maybe we should return it directly? That won't be helpful if the image has an annoying PIL mode like `F`, though. So maybe this function should really be called `coerce_into_rgb` (tying in to my previous comment about the semantics of the `aspil()` method).

Third, if the data array is two-dimensional — which is going to be the most common usage — the `array[..., :3]` slice will select only the first three columns of the image. The bulk of the array data won't get scaled correctly. Due to the existence of the `F16x3` image mode, I think this function will need to branch its behavior depending on whether
the array's shape looks like `(H, W)` or `(H, W, 3)`. The only `(H, W, 4)` mode that we have is RGBA.

But fourth, I think we also need to think about whether this function should return either RGB or RGBA, or choose one. It it can return RGBA, we'd need to initialize the alpha channel appropriately, setting it to zero for NaNs and the like. But WWT thumbnails must be delivered as RGB JPEGs. So for the proximate use case, we need to drop down the alpha channel eventually.

So, synthesizing all of that, I think that this function should actually explicitly be `coerce_to_rgb_pil()`, with logic to fill in all three channels equally for formats like F32. Non-finite pixels (`~np.isfinite(array)`, which includes NaNs) should be mapped to black. If there is already a `_pil` representation, it should be returned if it is in `RGB` mode, or if it is in `RGBA` mode, it should be downconverted sensibly (there's probably a PIL function do this?). Other PIL modes should be handled with the `asarray()` codepath.